ea-2021-04-ndvi-automation-brian-pinke.py

```diff
@@ -0,0 +1,790 @@
+                           columns=["site", "date", "mean_ndvi"])
+    ndvi_df["date"] = pd.to_datetime(ndvi_df["date"])
+    ndvi_df = ndvi_df.reset_index().set_index(["date"])
+
+    return ndvi_df
+
+
+# # The Data Used
+# * The same data used in task 1, task 2, and the figure
+# * The data is imported using et.get_data(ndvi_automation)
+# * This data is Landsat data from two sites:
+#     * San Joaquin Experimental Range / SJER
+#     * Harvard Forest / HARV
+# * SJER is a terrestrial NEON field site located approximately 40 km (25 mi.) north of Fresno, CA. 
+# * HARV is a terrestrial NEON field site located approximately 65 miles west of Boston, Massachusetts in the county of Worcester. 
+
+# In[6]:
+
+
+# Create dataframe of mean NDVI in this cell using the functions created above
+# Important: to use the ungraded tests below as a sanity check,
+# name your columns: mean_ndvi and site
+# Call the dataframe at the end of the cell so the tests run on it!
+# Be sure that the date column is an index of type date
+# HINT: the time series lessons may help you remember how to do this!
+
+# YOUR CODE HERE
+# Get a list of each directory
+path = os.path.join("ndvi-automation", "sites")
+
+# Get a list of both site directories
+sites = glob(path + "/*/")
+
+# Get the site name
+site_name = os.path.basename(os.path.normpath(sites[0]))
+
+# Open up the shapefile for clipping your landsat data to the study area
+vector_dir = os.path.join(sites[0],
+                          "vector")
+
+# Open crop boundary
+site_boundary_path = os.path.join(vector_dir,  site_name + "-crop.shp")
+crop_bound = gpd.read_file(site_boundary_path)
+
+landsat_dir = os.path.join(sites[0],
+                           "landsat-crop")
+
+# Path to all desired .tif files
+landsat_dirs = sorted(glob(os.path.join(landsat_dir, "LC08*")))
+
+# Select just a single directory and grab bands 4-5 from the directory
+adir = landsat_dirs[4]
+
+# Open bands
+band_paths = sorted(glob(os.path.join(adir, "*band*[4-5].tif")))
+
+# Open and clip the cloud mask layer
+# Cloud no data vals for Landsat 8 -
+vals = [328, 392, 840, 904, 1350, 352, 368, 416,
+        432, 480, 864, 880, 928, 944, 992, 480, 992]
+
+# Get cloud mask layer path
+qa_r = glob(os.path.join(adir, "*qa*"))
+
+# Clip the cloud mask layer
+cl_mask = rxr.open_rasterio(qa_r[0], masked=True).rio.clip(
+    crop_bound.geometry, from_disk=True).squeeze()
+
+# Open and clean bands
+all_bands = []
+for aband in band_paths:
+    cleaned_band = open_clean_bands(band_path=aband,
+                                    crop_bound=crop_bound,
+                                    valid_range=(0, 10000),
+                                    a_mask=cl_mask,
+                                    vals=vals)
+    all_bands.append(cleaned_band)
+
+
+# Calculate mean ndvi
+final_ndvi = calc_mndvi(all_bands)
+
+# Append site name, site date, ndvi mean to dir_list
+dir_list = temp_list(site_name=site_name, adir=adir, ndvi=final_ndvi)
```

Comment:
Good stuff, I like this being wrapped up in a function that can be reused between the tasks!

Reply:
Thanks! Its one of those small, yet easy things to do so I figured why not.